sonic-thermalctld/scripts/thermalctld

```diff
@@ -610,19 +614,18 @@ class TemperatureUpdater(logger.Logger):
                         self._refresh_temperature_status(sfp_name, thermal, thermal_index)
 
                 for psu_index, psu in enumerate(module.get_all_psus()):
-                    psu_name = '{} PSU {}'.format(module_name, psu_index + 1)
-                    for thermal_index, thermal in enumerate(psu.get_all_thermals()):
-                        if self.task_stopping_event.is_set():
-                            return
-
-                        available_thermals.add((thermal, psu_name, thermal_index))
-                        self._refresh_temperature_status(psu_name, thermal, thermal_index)
-
-
-            thermals_to_remove = self.module_thermals - available_thermals
-            self.module_thermals = available_thermals
-            for thermal, parent_name, thermal_index in thermals_to_remove:
-                self._remove_thermal_from_db(thermal, parent_name, thermal_index)
+                    if psu.get_presence() and psu.get_powergood_status():
```

Comment:
Please change this to trust the "present" signal only. PSU good or bad is not a criteria for temperature sensor readout.

Reply:
Updated.